File: gitdot-api-ts/src/resource/runner.ts
import { z } from "zod";

export const RunnerResource = z.object({
  id: z.string().uuid(),
  name: z.string(),
  owner_id: z.string().uuid(),
  owner_name: z.string(),
  owner_type: z.string(),
  last_verified: z.string().datetime().nullable(),
  created_at: z.string().datetime(),
});
export type RunnerResource = z.infer<typeof RunnerResource>;

export const RunnerTokenResource = z.object({
  token: z.string(),
});
export type RunnerTokenResource = z.infer<typeof RunnerTokenResource>;

File: gitdot-api/src/resource/runner.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(ApiResource, Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct RunnerResource {
    pub id: Uuid,
    pub name: String,
    pub owner_id: Uuid,
    pub owner_name: String,
    pub owner_type: String,
    pub last_active: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
}

#[derive(ApiResource, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct RunnerTokenResource {
    pub token: String,
}
